File: business/src/main/java/no/nav/sbl/sosialhjelp/soknad/SoknadUnderArbeidRepository.java
package no.nav.sbl.sosialhjelp.soknad;

import no.nav.sbl.sosialhjelp.domain.SoknadUnderArbeid;

public interface SoknadUnderArbeidRepository {

    Long opprettSoknad(SoknadUnderArbeid soknadUnderArbeid, String eier);
    SoknadUnderArbeid hentSoknad(Long soknadId, String eier);
    SoknadUnderArbeid hentSoknad(String behandlingsId, String eier);
    void oppdatereSoknadsdata(SoknadUnderArbeid soknadUnderArbeid, String eier);
    void slettSoknad(SoknadUnderArbeid soknadUnderArbeid, String eier);
}

File: src/main/kotlin/no/nav/sosialhjelp/soknad/db/repositories/soknadunderarbeid/SoknadUnderArbeidRepository.kt
package no.nav.sosialhjelp.soknad.db.repositories.soknadunderarbeid

import java.time.LocalDateTime
import java.time.temporal.ChronoUnit

interface SoknadUnderArbeidRepository {
    fun opprettSoknad(
        soknadUnderArbeid: SoknadUnderArbeid,
        eier: String,
    ): Long?

    fun hentSoknad(
        soknadId: Long,
        eier: String,
    ): SoknadUnderArbeid?

    fun hentSoknad(
        behandlingsId: String?,
        eier: String,
    ): SoknadUnderArbeid

    fun hentSoknadNullable(
        behandlingsId: String?,
        eier: String,
    ): SoknadUnderArbeid?

    fun oppdaterSoknadsdata(
        soknadUnderArbeid: SoknadUnderArbeid,
        eier: String,
        sistEndretDato: LocalDateTime = LocalDateTime.now().truncatedTo(ChronoUnit.SECONDS),
    )

    fun oppdaterInnsendingStatus(
        soknadUnderArbeid: SoknadUnderArbeid,
        eier: String,
    )

    fun slettSoknad(
        soknadUnderArbeid: SoknadUnderArbeid,
        eier: String,
    )
}
